lib/index.js

```diff
@@ -248,18 +249,22 @@ internals.finish = function (response, req, onEnd) {
 
         res.rawPayload = new Buffer(payloadBytes);
         res.payload = res.rawPayload.toString('utf8');
-        const headers = rest.toString().split(CRLF);
-        headers.forEach((header) => {
-
-            const headerParts = header.split(':');
-            if (headerParts.length === 2) {
-                response._headers[headerParts[0].trim().toLowerCase()] = headerParts[1].trim();
-            }
-        });
```

Comment:
You need to still use the headers at the tail of the payload. That's the proper place to get them from so that it truly simulates the headers sent. Just need to assign them to the right property instead of lumping them with the normal headers.